share/dkg/dkg_test.go

```diff
@@ -0,0 +1,616 @@
+package dkg
+
+import (
+	"crypto/rand"
+	"testing"
+
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/ed25519"
+	"github.com/dedis/crypto/random"
+	"github.com/dedis/crypto/share"
+	"github.com/dedis/crypto/share/vss"
+	"github.com/dedis/crypto/sign"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var suite = ed25519.NewAES128SHA256Ed25519(false)
+
+var nbParticipants = 7
+
+var partPubs []abstract.Point
+var partSec []abstract.Scalar
+
+var dkgs []*DistKeyGenerator
+
+func dkgGen() []*DistKeyGenerator {
+	dkgs := make([]*DistKeyGenerator, nbParticipants)
+	for i := 0; i < nbParticipants; i++ {
+		dkg, err := NewDistKeyGenerator(suite, partSec[i], partPubs, random.Stream, nbParticipants/2+1)
+		if err != nil {
+			panic(err)
+		}
+		dkgs[i] = dkg
+	}
+	return dkgs
+}
+
+func genPair() (abstract.Scalar, abstract.Point) {
+	sc := suite.Scalar().Pick(random.Stream)
+	return sc, suite.Point().Mul(nil, sc)
+}
+
+func randomBytes(n int) []byte {
+	var buff = make([]byte, n)
+	rand.Read(buff[:])
+	return buff
+}
+
+func init() {
```

Comment:
style: can we have `init` at the top, please? Just after the constants `suite` and `nbParticipants`.